App/App.cpp

```diff
@@ -1,28 +1,48 @@
-#include <stdio.h>
-#include <climits>
-#include "App.h"
-#include "SgxException.h"
-#include "EnclaveManager.h"
-
-long affineTransformation(unsigned long x, long targetRangeMin, long targetRangeMax, unsigned long sourceRangeMin, unsigned long sourceRangeMax) {
-	return ((double)x - (double)sourceRangeMin)*((double)targetRangeMax - (double)targetRangeMin) / ((double)sourceRangeMax - (double)sourceRangeMin) + (double)targetRangeMin;
-}
-
-int generateRandom(long min, long max, long *result) {
-	try {
-		EnclaveManager enclave;
-		unsigned long generated = enclave.generateRandom();
-
-		*result = affineTransformation(generated, min, max, 0, ULONG_MAX);
-		printf("Transforming random value from SGX [0, %lu] to [%ld, %ld]: %lu -> %ld \n", ULONG_MAX, min, max, generated, *result);
-	} catch (SgxException& e) {
-		printf("%s\n", e.what());
-		return -1;
-	}
-
-	return 0;
-}
-
-int main() {
-	return 0;
-}
+#include <stdio.h>
+#include <climits>
+#include "App.h"
+#include "SgxException.h"
+#include "EnclaveManager.h"
+
+long affineTransformation(unsigned long x, long targetRangeMin, long targetRangeMax, unsigned long sourceRangeMin, unsigned long sourceRangeMax) {
+	return ((double)x - (double)sourceRangeMin)*((double)targetRangeMax - (double)targetRangeMin) / ((double)sourceRangeMax - (double)sourceRangeMin) + (double)targetRangeMin;
```

Comment:
I can see 2 issues here:
1. too much casting to double, if you need to force floating point arithmetic (division in your case) then it's enough to cast only one of the arguments: `(double) (x - sourceRangeMin)*(targetRangeMax - targetRangeMin) / (sourceRangeMax - sourceRangeMin) + targetRangeMin`. It should be even faster because multiplication and subtraction will be done on integers.
2. You function wants to return `long` but you have there `double`. Currently it's being truncated to nearest integer so 5.61234 becomes 5. Is that what you want? Maybe you could be more explicit in this case and use one of the math functions: round, ceil or floor

Reply:
1 - ok
2 - yes, this is intentional. I will use floor to make it obvious, performance is not key there so it won't hurt